dwave/cloud/client.py

```diff
@@ -144,13 +171,36 @@ class Client(object):
     STATUS_FAILED = 'FAILED'
     STATUS_CANCELLED = 'CANCELLED'
 
-    # Default API endpoint
-    DEFAULT_API_ENDPOINT = 'https://cloud.dwavesys.com/sapi/'
-
     # Cases when multiple status flags qualify
     ANY_STATUS_ONGOING = [STATUS_IN_PROGRESS, STATUS_PENDING]
     ANY_STATUS_NO_RESULT = [STATUS_FAILED, STATUS_CANCELLED]
 
+    # Default API endpoint
+    DEFAULT_API_ENDPOINT = 'https://cloud.dwavesys.com/sapi/'
+
+    # Class-level defaults for all constructor and factory arguments
+    DEFAULTS = {
+        # factory only
+        'config_file': None,
+        'profile': None,
+        'client': 'base',
+        # constructor (and factory)
+        'endpoint': DEFAULT_API_ENDPOINT,
+        'token': None,
+        'solver': None,
+        'proxy': None,
+        'permissive_ssl': False,
+        'request_timeout': 60,
+        'polling_timeout': None,
+        'connection_close': False,
+        'headers': None,
+        'client_cert': None,
+        'client_cert_key': None,
+        # Poll back-off schedule defaults [sec]
```

Comment:
Why do these two get their own section?

Reply:
Starting a trend of options that will follow. :smiley: